src/ucp/api/ucp.h

```diff
@@ -3386,32 +3411,65 @@ ucp_atomic_fetch_nb(ucp_ep_h ep, ucp_atomic_fetch_op_t opcode,
 
 /**
  * @ingroup UCP_COMM
- * @brief Post an atomic fetch operation.
+ * @brief Post an atomic memory operation.
  *
- * This routine will post an atomic fetch operation to remote memory.
+ * This routine will post an atomic operation to remote memory.
  * The remote value is described by the combination of the remote
  * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
- * @a rkey.
- * The routine is non-blocking and therefore returns immediately. However the
- * actual atomic operation may be delayed. The atomic operation is not considered complete
- * until the values in remote and local memory are completed.
+ * @a rkey. The routine is non-blocking and therefore returns immediately.
+ * However the actual atomic operation may be delayed.
+ * If the reply buffer is not provided in @a param->reply_buffer, the request
+ * handle is not returned and the user must call @ref ucp_ep_flush_nb or
+ * @ref ucp_worker_flush_nb to guarantee that the remote value has been updated.
+ * Otherwise, if a fetch operation is being requested, the atomic
+ * operation is not considered complete until the values in both remote and
+ * local memory are completed. The result of the operation will be stored in the
+ * @a param->reply_buffer memory buffer. In the case of CSWAP the value in
+ * @a param->reply_buffer will be swapped into the @a remote_addr if the
+ * condition is true.
+ *
+ * @note    The user should not modify any part of the @a buffer (or also
+ *          @a param->reply_buffer for fetch operations), until the operation
+ *          completes.
+ * @note    Only ucp_dt_make_config(4) and ucp_dt_make_contig(8) are supported
+ *          in @a param->datatype, see @ref ucp_dt_make_contig.
+ * @warning Currently only fetch operations are supported by this routine.
+ *          Therefore @a param->reply_buffer and its corresponding flag
+ *          @ref UCP_OP_ATTR_FIELD_REPLY_BUFFER have to be defined in @a param.
```

Comment:
is this still relevant?

Reply:
yes, we support just fence semantic now

Reply:
do we plan to remove this restriction in the future?

Reply:
yes,
we can do in this PR by redirecting to atomic_post func and optimize later